modules/command-handler/src/application/aggregate.ts

```diff
@@ -0,0 +1,368 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+ Contributors
+ --------------
+ This is the official list (alphabetical ordering) of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+ * Modusbox
+ - Shashikant Hirugade <shashikant.hirugade@modusbox.com>
+ - Juan Correa <juancorrea@modusbox.com>
+
+ --------------
+ ******/
+
+ "use strict";
+
+ import {ILogger} from "@mojaloop/logging-bc-public-types-lib";
+ import * as uuid from "uuid";
+ import {
+    BulkTransferEntityAlreadyExistsError,
+     InvalidBulkTransferEntityIdTypeError,
+     NoSuchBulkTransferEntityError,
+ } from "./errors";
+ import {IBulkTransferEntityRepo} from "./infrastructure-interfaces/irepo";
+ import {BulkTransferEntity, BulkTransferRequest, BulkBatch} from "./types";
+ 
+ export class Aggregate {
+     // Properties received through the constructor.
+     private readonly logger: ILogger;
+     private readonly repo: IRepo;
+     // Other properties.
+ 
+     constructor(
+         logger: ILogger,
+         repo: IRepo
+     ) {
+         this.logger = logger;
+         this.repo = repo;
+     }
+
+     async init(): Promise<void> {
+		try {
+			await this.repo.init();
+		} catch (e: unknown) {
+			this.logger.fatal(e);
+			throw e; // No need to be specific.
+		}
+	}
+
+	async destroy(): Promise<void> {
+		await this.repo.destroy();
+	}
+
+     // TODO: Add update
+ 
+     //BulkTransferEntity
+     async createBulkTransferEntity(BulkTransferEntity: BulkTransferEntity): Promise<string> { // TODO: BulkTransferEntity or IBulkTransferEntity?
```

Comment:
These functions are probably going to match most of the Commands that appear on the Detail Design.

Nothing wrong with having these, but they may not end up being very useful since allot of the "crud" like operations will be handled by the Repo.